.github/workflows/update-mobile-repositories.yml

```diff
@@ -0,0 +1,358 @@
+name: Update All XMTP Repositories
+on:
+  workflow_dispatch:
+    inputs:
+      tag:
+        description: "Tag to use (e.g. swift-bindings-0.1.0.abc1234 or kotlin-bindings-0.1.0.abc1234)"
+        required: true
+      xmtp_ios_branch:
+        description: "xmtp-ios branch to base PR off of (default: main)"
+        required: false
+        default: "main"
+      xmtp_android_branch:
+        description: "xmtp-android branch to base PR off of (default: main)"
+        required: false
+        default: "main"
+      xmtp_react_native_branch:
+        description: "xmtp-react-native branch to base PR off of (default: main)"
+        required: false
+        default: "main"
+
+jobs:
+  get-version-info:
+    runs-on: ubuntu-latest
+    outputs:
+      version: ${{ steps.version_info.outputs.version }}
+      release_tag: ${{ steps.version_info.outputs.release_tag }}
+      binding_type: ${{ steps.version_info.outputs.binding_type }}
+    steps:
+      - name: Checkout libxmtp
+        uses: actions/checkout@v5
+        with:
+          ref: ${{ github.event_name == 'workflow_dispatch' && github.event.inputs.tag || github.ref }}
+
+      - name: Get version and SHA
+        id: version_info
+        run: |
+          # Get the original version from cargo metadata
```

Comment:
This is long enough it probably deserves to go into a shell script so that it gets spellcheck and proper syntax highlighting.